### ISSUE TITLE
Measure Bodies stage progress in gas instead of blocks

### ISSUE THREAD

**Issue by shekhirin** [paradigmxyz/reth#3061]
### Describe the feature

As TODO comment says: https://github.com/paradigmxyz/reth/blob/3aff8bcd1c938e7e650b5e14ec0d892200945750/crates/stages/src/stages/bodies.rs#L222-L232

**Comment by onbjerg** [comment]
Do we have any data to support that this would make sense? Afaik gas is more of a proxy for execution time (more gas = longer to execute because of i/o, cpu), not really the size. You can create blocks with a lot of gas consumption that are very small

**Comment by shekhirin** [comment]
> Afaik gas is more of a proxy for execution time, not really the size.

you're right. My rationale here is that gas would be a better (but still not ideal) approximation of progress than blocks, because otherwise all blocks are considered to be the same. Not sure what we can extract from the data we have that'd represent the Bodies downloader progress better.

**Comment by onbjerg** [comment]
Imo the current metric for bodies/headers is fine, usually they are not the bottlenecking stages overall anyway. Is there any specific reason you do not think so?

**Comment by shekhirin** [comment]
> usually they are not the bottlenecking stages overall anyway

good point, agree. Add adding the gas counting logic similar to Execution stage will complicate the code. @gakonst wdyt?

**Comment by gakonst** [comment]
I think it's fine as-is personally. Bodies are reasonably fast already.
